oscrypto/keys.py

```diff
@@ -21,4 +27,100 @@
     'parse_pkcs12',
     'parse_private',
     'parse_public',
+    "make_pkcs12",
 ]
+
+
+def make_pkcs12(key, cert, other_certs, password):
+    # XXX check if key matches cert
+    lkid = hashlib.sha1(cert.dump()).digest()
+    salt = rand_bytes(8)
+    eai = cms.EncryptionAlgorithm(
+        {
+            "algorithm": "pkcs12_sha1_rc2_40",
+            "parameters": {"salt": salt, "iterations": 2048},
+        }
+    )
+    content = [
+        pkcs12.SafeBag(
+            {
+                "bag_id": "cert_bag",
+                "bag_value": pkcs12.CertBag({"cert_id": "x509", "cert_value": cert}),
+                "bag_attributes": pkcs12.Attributes(
+                    [{"type": "local_key_id", "values": [core.OctetString(lkid)]}]
+                ),
+            }
+        )
+    ]
+    for cert in other_certs:
+        content.append(
+            pkcs12.SafeBag(
+                {
+                    "bag_id": "cert_bag",
+                    "bag_value": pkcs12.CertBag(
+                        {"cert_id": "x509", "cert_value": cert}
+                    ),
+                }
+            )
+        )
+    content = pkcs12.SafeContents(content).dump()
+    content = _encrypt_data(eai, content, password)
+    ced = cms.EncryptedData(
+        {
+            "version": "v0",
+            "encrypted_content_info": cms.EncryptedContentInfo(
+                {
+                    "content_type": "data",
+                    "content_encryption_algorithm": eai,
+                    "encrypted_content": content,
+                }
+            ),
+            "unprotected_attrs": None,
+        }
+    )
+
+    salt = rand_bytes(8)
+    eai = cms.EncryptionAlgorithm(
+        {
+            "algorithm": "pkcs12_sha1_tripledes_3key",
+            "parameters": {"salt": salt, "iterations": 2048},
+        }
+    )
+    content = _encrypt_data(eai, key.dump(), password)
+    eki = pkcs12.SafeBag(
+        {
+            "bag_id": "pkcs8_shrouded_key_bag",
+            "bag_value": pkcs12.EncryptedPrivateKeyInfo(
+                {"encryption_algorithm": eai, "encrypted_data": content}
+            ),
+            "bag_attributes": pkcs12.Attributes(
+                [{"type": "local_key_id", "values": [core.OctetString(lkid)]}]
+            ),
+        }
+    )
+    config = [
+        cms.ContentInfo({"content_type": "encrypted_data", "content": ced}),
+        cms.ContentInfo(
+            {"content_type": "data", "content": pkcs12.SafeContents([eki]).dump()}
+        ),
+    ]
+    content = pkcs12.AuthenticatedSafe(config).dump()
+    mac_salt = rand_bytes(8)
+    mac_algo = "sha1"
+    mac_iterations = 2048
```

Comment:
Some comments should be left here explaining why these parameter values were picked. I presume it has something to do with compatibility, so if so it would be good to know what browser/programs these are designed to work with.